File: MD.CMS.BusinessLogic.Core/DataAccess/Entities/ReportSchedulerAction.cs
using MD.Tools.BaseDataAccess.Core.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MD.CMS.BusinessLogic.Core.DataAccess.Entities
{
   public class ReportSchedulerAction:BaseEntity<long>
   {
       #region Attributes
       private long _schedulerId;
       private string _name;
       private string _authorId;
       private DateTime _dateCreated;
       private DateTime _dateEdited;
       private EnumAction _actionType;
       private string _options;
       private bool _isActive;
       #endregion

       #region Properties
       public long SchedulerId
       {
           get { return _schedulerId; }
           set { _schedulerId = value; }
       }
       public string Name
       {
           get { return _name; }
           set { _name = value; }
       }

       public string AuthorId
       {
           get { return _authorId; }
           set { _authorId = value; }
       }
       public DateTime DateCreated
       {
           get { return _dateCreated; }
           set { _dateCreated = value; }
       }
       public DateTime DateEdited
       {
           get { return _dateEdited; }
           set { _dateEdited = value; }
       }
       public EnumAction ActionType
       {
           get { return _actionType; }
           set { _actionType = value; }
       }
       public string Options
       {
           get { return _options; }
           set { _options = value; }
       }
       public bool IsActive
       {
           get { return _isActive; }
           set { _isActive = value; }
       }
       #endregion

       #region Enums
       public enum EnumAction : int
       {
           SaveToDisk = 1,
           Email = 2,
       }
     
       #endregion
    }
}

File: MD.CMS.Administration/MD.CMS.Administration.Core.Web/scripts/businessLogic/dataAccess/entities/reportSchedulerAction.ts
/// <reference path="./base/iBaseEntity.ts" />
/// <reference path="./base/baseEntity.ts" />

namespace mdBusinessLogic.dataAccess.entities {
    export class reportSchedulerAction extends base.BaseEntity implements base.IBaseEntity<reportSchedulerAction> {
        public SchedulerId: number;
        public Name: string;
        public AuthorId: number;
        public DateCreated: Date;
        public DateEdited: Date;
        public ActionType: number;
        public Options: string;
        public IsActive: boolean;

        constructor(obj?: reportSchedulerAction) {
            super(obj);
            this.SchedulerId = 0;
            this.Name = '';
            this.AuthorId = 0;
            this.DateCreated = new Date();
            this.DateEdited = null;
            this.ActionType = 0;
            this.Options = '';
            this.IsActive = false;
            if (obj != undefined && obj != null) {
                this.construct(obj);
            }
        }

        construct(data: any): void {
            super.construct(data);
            this.SchedulerId = this.getValue<number>(data, "SchedulerId", 0);
            this.Name = this.getValue<string>(data, "Name", '');
            this.AuthorId = this.getValue<number>(data, "AuthorId", 0);
            this.DateCreated = this.getValue<Date>(data, "DateCreated", new Date());
            this.DateEdited = this.getValue<Date>(data, "DateEdited", null);
            this.ActionType = this.getValue<number>(data, "ActionType", 0);
            this.Options = this.getValue<string>(data, "Options", '');
            this.IsActive = this.getValue<boolean>(data, "IsActive", false);
        }

        public clone(): reportSchedulerAction {
            return new reportSchedulerAction(this);
        }

    }
}